### PR TITLE
Allow for manually specifying the url to the API server

### PR DESCRIPTION
**Pull request by bobsilverberg**
Remove unittestzero and switch to native asserts
Update readme to reflect new usage
Update pyfxa requirement to latest version
Update Travis config to only run flake8 check

### CODE REVIEW

**Comment by bobsilverberg** [comment]
@davehunt r?

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+* Change FxATestAccount constructor to accept the url to the FxA API server
+  * This is a **breaking change**
+
+1.2
+---
+* Update required version of PyFxA in setup.py
+
+1.1
+---
+* Update required version of PyFxA in requirements.txt
```
Comment (davehunt) [inline CHANGELOG.md]: Do we need a change log? If so, I think the best place for this to be listed is PyPI but this file isn't currently displayed there. Probably a future enhancement but I'd recommend switching the README to be displayed on PyPI and include the change log there. See an example of pulling in the README contents for PyPI here: https://github.com/davehunt/pytest-html/blob/fa3cf426036a2cdc3a5b87e7e604937bc0ee88a2/setup.py#L6

Comment (davehunt) [inline CHANGELOG.md]: Oh, and for 1.1 and 1.2 I'd reword to 'Update required version of PyFxA to x.x' rather than mention the file that maintains the dependency.

File: fxapom/fxapom.py [hunk 2]
```diff
@@ -27,11 +32,12 @@ class FxATestAccount:
 
     password = ''.join([random.choice(string.letters) for i in range(8)])
 
-    def __init__(self, base_url=None):
-        if base_url and '-dev.allizom' in base_url:
-            self.fxa_url = 'https://stable.dev.lcip.org/auth/'
-        else:
-            self.fxa_url = 'https://api.accounts.firefox.com/'
+    def __init__(self, api_host_url=None):
```
Comment (davehunt) [inline fxapom/fxapom.py]: I would name the argument simply 'url' and directly make the default DEV_URL.

File: fxapom/pages/page.py [hunk 3]
```diff
@@ -87,6 +86,6 @@ def wait_for_element_not_present(self, *locator):
             WebDriverWait(self.selenium, self.timeout).until(lambda s: len(self.find_elements(*locator)) < 1)
             return True
         except TimeoutException:
-            Assert.fail(TimeoutException)
+            raise TimeoutException
```
Comment (davehunt) [inline fxapom/pages/page.py]: This is not exactly the same. If we want to just raise the TimeoutException then we shouldn't catch it. If we want to fail here with an assertion error then we should raise an AssertionError.

File: readme.md [hunk 4]
```diff
@@ -13,13 +13,30 @@ as well as a set of page objects that can be used to interact with Firefox Accou
...
+Example of creating an account on the development environment, specifying the `DEV_URL`:
+```python
+from fxapom.fxapom import DEV_URL, FxATestAccount
+acct = FxATestAccount(DEV_URL).create_account()
+```
+
+Example of creating an account on the production environment, specifying the `PROD_URL`:
+```python
+from fxapom.fxapom import FxATestAccount, PROD_URL
+acct = FxATestAccount(PROD_URL).create_account()
```
Comment (davehunt) [inline readme.md]: Do we really need this example?

File: tests/test_create_account.py [hunk 5]
```diff
@@ -4,26 +4,29 @@
...
 class TestCreateAccount(object):
 
-    def test_create_new_account_on_dev(self):
-        acct = FxATestAccount(base_url='https://www-dev.allizom.org').create_account()
-        Assert.true(acct.is_verified)
-        Assert.equal('https://stable.dev.lcip.org/auth/', acct.fxa_url)
+    def test_create_new_account_on_dev_by_default(self):
+        acct = FxATestAccount().create_account()
+        assert acct.is_verified
+        assert 'https://stable.dev.lcip.org/auth/' == acct.fxa_url
```
Comment (davehunt) [inline tests/test_create_account.py]: Do we need to create an account here? It seems we're just testing the default value of the url argument. We could do this by just writing `assert DEV_URL == FxATestAccount().fxa_url`. This leads me to wonder why we can create an account object without creating an actual account - is there a use case for that?

File: tests/test_create_account.py [hunk 6]
```diff
@@ -4,26 +4,29 @@
...
-        Assert.equal('https://api.accounts.firefox.com/', acct.fxa_url)
+    def test_create_new_account_on_dev_explicitly(self):
+        acct = FxATestAccount(DEV_URL).create_account()
+        assert acct.is_verified
+        assert 'https://stable.dev.lcip.org/auth/' == acct.fxa_url
+
+    def test_create_new_account_on_prod(self):
+        acct = FxATestAccount(PROD_URL).create_account()
+        assert acct.is_verified
+        assert 'https://api.accounts.firefox.com/' == acct.fxa_url
```
Comment (davehunt) [inline tests/test_create_account.py]: I think we need clarification of what we're creating here - should we be concerned that we're creating 'production' accounts?

File: CHANGELOG.md (hunk 1 above)
Comment (bobsilverberg) [inline CHANGELOG.md]: Fixed both of these in the next commit.

File: fxapom/fxapom.py (hunk 2 above)
Comment (bobsilverberg) [inline fxapom/fxapom.py]: Done

File: fxapom/pages/page.py (hunk 3 above)
Comment (bobsilverberg) [inline fxapom/pages/page.py]: Fixed

File: readme.md (hunk 4 above)
Comment (bobsilverberg) [inline readme.md]: I suppose not.

File: tests/test_create_account.py (hunk 5 above)
Comment (bobsilverberg) [inline tests/test_create_account.py]: I think this is related to the discussion we had about the weirdness of the code. I will change the test to not create an account, and we should also revisit the code in a separate issue to see if it can be improved.

File: tests/test_create_account.py (hunk 6 above)
Comment (bobsilverberg) [inline tests/test_create_account.py]: I think we discussed this today, but I need to double check about exactly what a 'test production' account is.

**Comment by bobsilverberg** [comment]
Thanks for the review @davehunt. I have addressed your comments in the latest commit.

**Comment by davehunt** [comment]
Looks good, my comments do not block merging this, but I've added this repo to our potential hack sessions for next week. :smile:

**Comment by bobsilverberg** [comment]
Thanks for the reviews, @davehunt. I will squash the commits and merge this so we have a good base to hack on.
